### ISSUE TITLE
Parliamentary coalition mechanic

### ISSUE THREAD

**Issue by frnsys** [frnsys/half_earth#67]
Right now "coalitions" just means good or bad relationships with NPC factions and getting a bonus if they're your ally.

It would be nice to add more depth to this, to really capture the political aspect of planning. Some projects could be locked unless a sufficient coalition is acquired (say a majority of the NPC factions), or unless you have a specific ally (e.g. the Malthusian), or some combination of both.

- [x] Update planning page to show parliament. This should show the seats, what faction occupies them (color-coded, tap to learn more about that faction), with a highlight (border color or something) to indicate if they're in your coalition. Should list out the factions in your coalition and your % control of parliament
- [x] Procedure to determine faction parliament seats (all start even at first? then change based on your "relationship" with them)
- [x] Blocked projects (diff than locked: visible, but can't be implemented until conditions are met). Two types of conditions: majority in parliament and faction X in your coalition (can this just use the existing supporter/opposer system?)
- [x] Show seat changes in the report view

**Comment by frnsys** [comment]
We change the "Coalition" page to be "Parliament", and instead of the current layout/relationship system, we show the seats of parliament and how many seats each faction has. We could probably use the existing relationship system for this but with some tweaks.

**Comment by frnsys** [comment]
Coalition is still based on relationships, using the existing system. If you have a good relationship with the faction (say >=5/6), they're in your coalition and their seats count towards your total count

The faction seats could work like so:

1. Each faction has a number of support points `p`
2. Look at "new" projects (finished within the last 10 years) and count their supporters and opposers. The total count of supporters and opposers is `F`
3. Look at how contentedness changes in the past planning cycle, say it's `c`
4. For each supporter, update `p += c/F`, and for each opposer, update `p -= c/F`
5. Calculate total points `p` across all factions as `P`
6. For each faction, update their seats proportion `s` to `s = p/P`

Basically this is a way for people to assign "blame" to factions for the current state of the world, based on what projects they're associated with.

**Comment by frnsys** [comment]
How project blocking could work:

Say a project has `n` opposers and `m` of them are in your coalition. You need `m/1+m` majority to implement the project. So the more opposers there are, the more of a majority you will need.
